ctapipe/io/datawriter.py

```diff
@@ -797,3 +801,24 @@ def _write_context_metadata_headers(self):
             context_dict[key] = value
 
         meta.write_to_hdf5(context_dict, self._writer.h5file)
+
+    def _write_atmosphere_profile(self, path):
+        """
+        write atmosphere profiles if they are in a tabular format
+
+        Parameters
+        ----------
+        path: str
+            path in the HDF5 file where to place the profile
+
+        """
+
+        profile = self.event_source.atmosphere_density_profile
+
+        if profile and hasattr(profile, "table"):
```

Comment:
This just silently fails does not write out the  `profile is not None` but `not hasattr(profile, "table")`

Reply:
And only profiles with tables  can be written here (e.g. you can't write an ExponentialAtmosphereDensityProfile currently, though I suppose that could be implemented in the future)

Reply:
I suppose a warning that the type of profile is not serializable could be provided.

Reply:
The exponential profile is a table also, right? A table of parameters